backend/src/services/auth.rs

```diff
@@ -37,10 +39,31 @@ impl AuthService {
             .map_err(HashPasswordFailed)?
             .to_string();
 
-        let uuid = Uuid::new_v4();
-        users::Model::create_with_password(database, uuid, password_hash, &credentials).await?;
+        let confirmation_code =
+            users::Model::create_with_password(&app_state.database, password_hash, &credentials)
+                .await?;
+        let mut confirmation_link = "".to_string();
+        confirmation_link.push_str(&EnvConfig::get().email_confirm_backend_url.clone());
+        confirmation_link.push('/');
+        confirmation_link.push_str(&confirmation_code);
```

Comment:
SRP, single function, single responsibility. For example:
- create new private function `confirmation_link` that creates confirmation link